models/m_payment_mollie_api.erl

```diff
@@ -137,65 +147,240 @@ payment_url(MollieId) ->
-                    update_payment_status(PaymentId, Status, Context),
-                    case proplists:get_value(recurring, Payment) of
-                        true ->
-                            create_subscription(Payment, Context);
-                        false ->
-                            ok
-                    end;
-                {error, Error} ->
-                    %% Log an error with the payment
+            case proplists:get_value(psp_module, Payment) of
+                mod_payment_mollie ->
+                    % Fetch the status from Mollie
+                    PaymentId = proplists:get_value(id, Payment),
                     m_payment_log:log(
-                      PaymentId,
-                      "ERROR",
-                      [
-                       {psp_module, mod_payment_mollie},
-                       {description, "API Error fetching status from Mollie"},
-                       {request_result, Error}
-                      ],
-                      Context),
-                    lager:error("API error creating mollie payment for #~p: ~p", [PaymentId, Error]),
-                    Error
+                        PaymentId,
+                        <<"WEBHOOK">>,
+                        [
+                           {psp_module, mod_payment_mollie},
+                           {description, "New webhook payment info"},
+                           {payment, JSON}
+                        ],
+                        Context),
+                    handle_new_payment(PaymentId, Payment, JSON, Context);
+                PSP ->
+                    lager:error("Payment PSP Mollie webhook call for unknown PSP ~p / ~p: ~p", [PaymentNr, ExtId, PSP]),
+                    {error, notfound}
             end;
         {error, notfound} ->
-            lager:error("Payment PSP Mollie webhook call with unknown id ~p", [ExtId]),
+            lager:error("Payment PSP Mollie webhook call with unknown id ~p / ~p", [PaymentNr, ExtId]),
             {error, notfound};
         {error, _} = Error ->
-            lager:error("Payment PSP Mollie webhook call with id ~p, fetching payment error: ~p", [ExtId, Error]),
+            lager:error("Payment PSP Mollie webhook call with id ~p / ~p, fetching payment error: ~p", [PaymentNr, ExtId, Error]),
+            Error
+    end;
+handle_missed_recurrent(_NonRecurringPayment, _Context) ->
+    ok.
+
+
+%% @doc Pull the payment status from Mollie.
+-spec payment_sync(binary() | integer(), binary(), z:context()) -> ok | {error, notfound|term()}.
+payment_sync(PaymentNrOrId, ExtId, Context) ->
+    case m_payment:get(PaymentNrOrId, Context) of
+        {ok, Payment} ->
+            case proplists:get_value(psp_module, Payment) of
+                mod_payment_mollie ->
+                    % Fetch the status from Mollie
+                    PaymentId = proplists:get_value(id, Payment),
+                    lager:info("Payment PSP Mollie webhook call for payment #~p", [PaymentId]),
```

Comment:
Is this a webhook (callback) call or just a request to the Mollie payment api?

Reply:
This routine is (at the moment) only called in response to a web hook call by Mollie.